File: OpenCV4 基本プログラミング/OpenCV4 基本プログラミング/src/flip.cpp
//
//  flip.cpp
//  OpenCV4 基本プログラミング
//
//  Created by 樅山柚子 on 2020/03/17.
//  Copyright © 2020 p_as_ta. All rights reserved.
//

#include <opencv2/opencv.hpp>

using namespace std;
using namespace cv;

void my_flip(int argc, char* argv[]) {
    try {
        if (argc < 3) {
            throw "few parameter, e.g. <file name> <flip code>.";
        }
        Mat src = imread(argv[1]);
        int flipCode = stoi(argv[2]);
        
        if(src.empty()) {
            throw "faild open file.";
        }
        
        Mat dst;
        flip(src, dst, flipCode);
        
        imshow("src", src);
        imshow("dst", dst);
        
        waitKey(0);
    } catch (const char* str) {
        cerr << str << endl;
    }
}

File: OpenCV4-BasicProgramming/src/Chapter4/flip.py
import cv2


def main(path: str, flip_code: int):
    src = cv2.imread(path)
    if len(src) == 0:
        raise NameError("failed open file.")
    dst = cv2.flip(src, flip_code)
    cv2.imshow("src", src)
    cv2.imshow("dst", dst)
    cv2.waitKey(0)
    return


if __name__ == '__main__':
    main()
